repose-aggregator/core/repose-core/src/main/scala/org/openrepose/commons/utils/classloader/EarClassProvider.scala

```diff
@@ -137,18 +138,31 @@ class EarClassProvider(earFile: File, val outputDir: File) {
       val zis = new ZipInputStream(new FileInputStream(earFile))
 
       val buffer = new Array[Byte](1024)
-      Stream.continually(zis.getNextEntry).
-        takeWhile(_ != null).foreach(entry =>
-        //Unpack the entry
+      Stream.continually(zis.getNextEntry).takeWhile(_ != null) foreach { entry =>
+        val entryFile = new File(outputDir, entry.getName)
         if (entry.isDirectory) {
-          new File(outputDir, entry.getName).mkdir()
+          entryFile.mkdir()
         } else {
-          val outFile = new File(outputDir, entry.getName)
-          val ofs = new FileOutputStream(outFile)
-          Stream.continually(zis.read(buffer)).takeWhile(_ != -1).foreach(count => ofs.write(buffer, 0, count))
-          ofs.close()
+          val ofs = new FileOutputStream(entryFile)
```

Comment:
Why ofs instead of fos?

Reply:
Well, I just kept the name already given to the variable to minimize changes. I'm not sure why it was given that name in the first place. I can change it, if you'd like.